CONTRIBUTING.md

```diff
@@ -0,0 +1,44 @@
+# Contributing
+
+Thanks for contributing to UTop!
+
+## Setting up local switches
+
+Create a `dune-workspace.multi` file with this contents:
+
+```
+(lang dune 1.0)
+(context (opam (switch utop-412)))
+(context (opam (switch utop-413)))
+(context (opam (switch utop-414)))
+(context (opam (switch utop-500)))
+(context (opam (switch utop-510)))
+(context (opam (switch utop-520)))
+```
+
+Run `make create-switches` to create all the local switches.
+
+Now you can run `dune` with the `--workspace dev-workspace.multi` flag to run
+the same command across all the workspaces.
+
+## Compatibility Across Versions
+
+Some code will be different from one version of OCaml to the next. If you find
+some that does, please abstract it away using the `UTop_compat` module.
+
+For example, the `Load_path.get_paths ()` function has changed recently to
+return a record with shape `{ visible: string list; hidden: string list }`, but
+this function used to return a single `string list`.
+
+Defining this function using pre-processor macros allows us to give the same
+function two different bodies on different version of the language. Note that
+they must have the exact same type signature.
```

Comment:
Not sure what you want me to do here :P – I'm happy rewriting/removing parts of that file, but I meant to write something that'd lead other folks to an easy time contributing. 

In my case, seeing UTop_compat as having a fixed interface made it easier to modify things in it, because I had a contract to comply to.

I understand there's nuance here, and that whatever signature the UTop_compat has is find as long as it fits the usage, but then I have to also consider and write about this, and for someone getting started this nuance seemed unnecessary.

Agin, happy to rewrite! Just let me know what would make sense to you (or if I missed something, what I misssed) 🙌🏼